### PR TITLE
Sol-monitors: Fix infinite loop while iterating over a sol-monitor callbacks V2.

### PR DESCRIPTION
**Pull request by cabelitos**
Changes since v1:
* Added a test

### CODE REVIEW

File: src/test/test-monitors.c [hunk 1]
```diff
@@ -354,5 +354,39 @@ custom_entry_type(void)
...
+infinite_loop_cb(const void *data)
+{
+    static uint8_t counter = 0;
+    struct sol_monitors *ms = (struct sol_monitors *)data;
+    void *v;
+
+    counter++;
+
+    if (counter != 1)
+        ASSERT(1 == 2);
```
Comment (barbieri) [inline src/test/test-monitors.c]: better to use `ASSERT_INT` (we have a macro to asser 2 integers are the same)

File: src/test/test-monitors.c [hunk 2]
```diff
@@ -354,5 +354,39 @@ custom_entry_type(void)
     sol_monitors_clear(ms);
 }
 
+DEFINE_TEST(infinite_loop_test);
+
+static void
+infinite_loop_cb(const void *data)
+{
+    static uint8_t counter = 0;
+    struct sol_monitors *ms = (struct sol_monitors *)data;
```
Comment (barbieri) [inline src/test/test-monitors.c]: no need to cast... and the function should be `(void *data)`, no const in there.

**Comment by barbieri** [comment]
just minor issues, you can push directly after fix

File: src/test/test-monitors.c (hunk 1 above)
Comment (cabelitos) [inline src/test/test-monitors.c]: You're right. I could have written ASSERT_INT_EQ(counter, 1)

File: src/test/test-monitors.c (hunk 2 above)
Comment (cabelitos) [inline src/test/test-monitors.c]: Actually sol_monitors_cb_t requires a const void * as argument. If I change here to void *, I will have to cast infinite_loop_cb when calling sol_monitors_append(). Either way I will need to cast something.

**Comment by cabelitos** [comment]
merged.
Thanks for the review.

**Pull request closed without merging**